src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -57,6 +57,23 @@ def wait_for_topics(admin_client: AdminClient, topics: List[str], timeout: int =
                 )
 
 
+def auto_create_topics(topics: List[str]):
+    """If configured to do so, create topics and make sure that they exist
+
+    topics must be from the same cluster.
+    """
+    if settings.KAFKA_CONSUMER_AUTO_CREATE_TOPICS:
+        cluster_names = {settings.KAFKA_TOPICS[topic]["cluster"] for topic in topics}
+        assert len(cluster_names) == 1
+        # This is required for confluent-kafka>=1.5.0, otherwise the topics will
+        # not be automatically created.
+        conf = kafka_config.get_kafka_admin_cluster_options(
+            cluster_names.pop(), override_params={"allow.auto.create.topics": "true"}
+        )
+        admin_client = AdminClient(conf)
+        wait_for_topics(admin_client, topics)
```

Comment:
I think it's ok to keep the if-check inside the function, but perhaps the `auto_` prefix in the function name is redundant (since we're basically implementing the "auto" part here). Alternatively, I think nobody will mind if we call it `auto_create_topics_if_configured` or something.
To make the outcome more obvious, you can also add a an "else" here, where you e.g. warn that the topics were not created because reasons.

Reply:
👍  renamed the function. I'm hesitant to log a warning here, because that would create noise in production even though it's desired behavior: https://github.com/getsentry/getsentry/blob/4f1dca7afa22edff8a8896ea3cfe8fc3c372c506/getsentry/conf/settings/prod.py#L519